### PR TITLE
Fix useFind in SSR: check for server Cursor

### PR DESCRIPTION
**Pull request by edemaine**
Fixes #334, as described in [this comment](https://github.com/meteor/react-packages/issues/344#issuecomment-1012334600).

Before this PR, server-side `useFind` printed a warning and returned `[]`. After this PR, the warning is gone and the correct `cursor.find()` results are returned.

### CODE REVIEW

**Comment by CaptainN** [comment]
Thanks so much for taking the time to produce a PR! I'll review and get back to you soon.

**Comment by CaptainN** [comment]
I actually think this is kind of important to fix and ship. cc: @StorytellerCZ

**Review by StorytellerCZ — APPROVED** [review]

File: packages/react-meteor-data/useFind.ts [hunk 1]
```diff
@@ -42,8 +42,16 @@ const useFindReducer = <T>(data: T[], action: useFindActions<T>): T[] => {
...
+// Check for valid Cursor or null.
+// On client, we should have a Mongo.Cursor (defined in
+// https://github.com/meteor/meteor/blob/devel/packages/minimongo/cursor.js and
+// https://github.com/meteor/meteor/blob/devel/packages/mongo/collection.js).
+// On server, however, we instead get a private Cursor type from
+// https://github.com/meteor/meteor/blob/devel/packages/mongo/mongo_driver.js
+// which has fields _mongo and _cursorDescription.
 const checkCursor = <T>(cursor: Mongo.Cursor<T> | undefined | null) => {
-  if (cursor !== null && cursor !== undefined && !(cursor instanceof Mongo.Cursor)) {
+  if (cursor != null && !(cursor instanceof Mongo.Cursor) &&
```
Comment (renanccastro) [inline packages/react-meteor-data/useFind.ts:53]: Shouldn't be enough checking for Meteor.isServer?

Comment (edemaine) [reply packages/react-meteor-data/useFind.ts:53]: The intent here is to print a warning if the argument is not an actual Meteor/Mongo cursor. I don't see how checking `Meteor.isServer` would suffice for that.

Comment (CaptainN) [reply packages/react-meteor-data/useFind.ts]: I addressed this. It's good to go.

File: packages/react-meteor-data/useFind.ts [hunk 2]
```diff
@@ -121,9 +129,7 @@ const useFindServer = <T = any>(factory: () => Mongo.Cursor<T> | undefined | nul
   Tracker.nonreactive(() => {
     const cursor = factory()
     if (Meteor.isDevelopment) checkCursor(cursor)
-    return (cursor instanceof Mongo.Cursor)
-      ? cursor.fetch()
-      : null
+    return cursor ? cursor.fetch() : null
```
[the remark this answers is not in the document]
Comment (CaptainN) [reply packages/react-meteor-data/useFind.ts]: I addressed this. Should be good to go.

**Comment by CaptainN** [comment]
This is good to go. I'm not sure why the CLA thing is failing though.

**Review by CaptainN — APPROVED** [review]
There were some minor issues with some of the new checks, but I changed what needed changing. Should be good to go.

File: packages/react-meteor-data/useFind.ts (hunk 1 above)
Comment (edemaine) [reply packages/react-meteor-data/useFind.ts]: Not a big deal, but `undefined` and `null` are the only two values that are `== null`. Reverting is also fine though; probably clearer not to use `==`.

**Review by edemaine — COMMENTED** [review]
I'm not a fan of how `cursor?.fetch?.() || null` transforms any falsey return value of `fetch` into `null`. This feels a bit magical; maybe it hides bugs? (in non-TypeScript code) 

`cursor?.fetch?.() ?? null` seems better; then only `undefined` gets remapped.

Actually, why not just `cursor?.fetch?.()`? This returns `undefined` if `cursor` or `cursor.fetch` are undefined/uncallable. Ah, but that's not isomorphic with the client. (But neither is the current node or the suggestion above, if `fetch` returns `undefined`.)

**Comment by CaptainN** [comment]
@edemaine Great point. To be completely honest, I just wasn't sure Meteor's build pipeline supports `??` but yeah, it should totally use that. Feel free to make the change, or I can do it later. Thanks!

**Comment by denihs** [comment]
Hey @CaptainN and @edemaine, as soon as this last matter is figured out, let me know, and I'll release these changes. :palms_up_together:

File: packages/react-meteor-data/useFind.ts (hunk 1 above)
Comment (CaptainN) [reply packages/react-meteor-data/useFind.ts]: Ah the vagaries of Javascript's type marshalling. It's probably a bit clearer with explicit checks, but that's good to know, and now that I do know, I don't have a strong opinion either way.

**Comment by denihs** [comment]
Version 2.5.1 is now up :tada:

**Pull request merged**